--- 1/sh/macro.c
+++ 1/sh/macro.c
@@ -322,6 +322,7 @@
 		if ((c = readwc(), dolchar(c))) {
 			struct namnod *n = (struct namnod *)NIL;
 			int		dolg = 0;
+			int		isg = 1;
 			int		vsize = -1;
 			BOOL		bra;
 			BOOL		nulflg;
@@ -362,6 +363,7 @@
 						atflag = 1;
 					}
 					dolg = 1;
+					isg++;
 					c = 1;
 				} else if (digit(c)) {
 					c -= '0';
@@ -622,7 +624,7 @@
 						error(badsub);
 					}
 				}
-			} else if (flags & setflg) {
+			} else if ((flags & setflg) && isg == 0) {
 				failed(id, unset);
 			}
 			goto retry;
