--- 1/work/usr/src/nv/conftest.sh
+++ 1/work/usr/src/nv/conftest.sh
@@ -651,7 +651,7 @@
             # a match() member.
             #
             echo "$CONFTEST_PREAMBLE
-            #include <acpi/acpi_bus.h>
+            #include <linux/acpi.h>
             int conftest_acpi_device_ops(void) {
                 return offsetof(struct acpi_device_ops, match);
             }" > conftest$$.c
@@ -893,7 +893,7 @@
             #
 
             echo "$CONFTEST_PREAMBLE
-            #include <acpi/acpi_bus.h>
+            #include <linux/acpi.h>
             acpi_status acpi_evaluate_integer(acpi_handle h, acpi_string s,
                 struct acpi_object_list *l, unsigned long long *d) {
                 return AE_OK;
@@ -910,7 +910,7 @@
             fi
 
             echo "$CONFTEST_PREAMBLE
-            #include <acpi/acpi_bus.h>
+            #include <linux/acpi.h>
             acpi_status acpi_evaluate_integer(acpi_handle h, acpi_string s,
                 struct acpi_object_list *l, unsigned long *d) {
                 return AE_OK;
@@ -958,7 +958,7 @@
             fi
 
             echo "$CONFTEST_PREAMBLE
-            #include <acpi/acpi.h>
+            #include <linux/acpi.h>
             void conftest_acpi_walk_namespace(void) {
                 acpi_walk_namespace(0, NULL, 0, NULL, NULL, NULL, NULL);
             }" > conftest$$.c
@@ -974,7 +974,7 @@
             fi
 
             echo "$CONFTEST_PREAMBLE
-            #include <acpi/acpi.h>
+            #include <linux/acpi.h>
             void conftest_acpi_walk_namespace(void) {
                 acpi_walk_namespace(0, NULL, 0, NULL, NULL, NULL);
             }" > conftest$$.c
@@ -1043,7 +1043,7 @@
                 echo "#define NV_ACPI_OS_WAIT_EVENTS_COMPLETE_ARGUMENT_COUNT 0" >> conftest.h
                 return
             else
-                echo "#error acpi_os_wait_events_complete() conftest failed!" >> conftest.h
+                : echo "#error acpi_os_wait_events_complete() conftest failed!" >> conftest.h
             fi
         ;;
 
