diff -urN foo2zjs.orig/icc2ps/Makefile foo2zjs/icc2ps/Makefile
--- foo2zjs.orig/icc2ps/Makefile	2007-11-27 22:13:53.000000000 +0100
+++ foo2zjs/icc2ps/Makefile	2008-12-02 13:51:12.000000000 +0100
@@ -1,11 +1,11 @@
 UNAME := $(shell uname)
-PREFIX=	/usr
+PREFIX=	$(DESTDIR)/usr
 BIN=	$(PREFIX)/bin
 SRC=	icc2ps.c xgetopt.c
 LIB=	cmscam97.c cmscnvrt.c cmserr.c cmsgamma.c cmsgmt.c cmsintrp.c cmsio1.c \
 	cmslut.c cmsmatsh.c cmsmtrx.c cmsnamed.c cmspack.c cmspcs.c cmsps2.c \
 	cmssamp.c cmswtpnt.c cmsxform.c cmsio0.c cmsvirt.c
-CFLAGS=	-O3
+CFLAGS?=	-O3
 INSTALL=install
 ifeq ($(UNAME),SunOS)
     INSTALL=/usr/ucb/install
diff -urN foo2zjs.orig/Makefile foo2zjs/Makefile
--- foo2zjs.orig/Makefile	2008-09-23 15:56:20.000000000 +0200
+++ foo2zjs/Makefile	2008-12-02 14:00:42.000000000 +0100
@@ -251,7 +251,7 @@
 endif
 
 # Compiler flags
-CFLAGS +=	-O2 -Wall
+CFLAGS ?=	-O2 -Wall
 #CFLAGS +=	-g
 
 #
@@ -543,14 +543,15 @@
 UDEVBIN=$(DESTDIR)/bin/
 
 install-prog:
+	[ -d $(BIN) ] || install -d -m 755 $(BIN)/
+	[ -d $(DESTDIR)/bin/ ] || install -d -m 755 $(DESTDIR)/bin/
 	#
 	# Install driver, wrapper, and development tools
 	#
 	$(INSTALL) -d $(BIN)
 	$(INSTALL) -c $(PROGS) $(SHELLS) $(BIN)/
 	if [ "$(BINPROGS)" != "" ]; then \
-	    $(INSTALL) -d $(UDEVBIN); \
-	    $(INSTALL) -c $(BINPROGS) $(UDEVBIN); \
+		install -c $(BINPROGS) $(DESTDIR)/bin/; \
 	fi
 	#
 	# Install gamma correction files.  These are just templates,
@@ -570,6 +571,7 @@
 	#
 	# Remove obsolete foomatic database files from previous versions
 	#
+	mkdir -p $(FOODB)/opt/
 	rm -f $(FOODB)/opt/foo2zjs-Media.xml
 	rm -f $(FOODB)/opt/foo2zjs-PaperSize.xml
 	rm -f $(FOODB)/opt/foo2zjs-Source.xml
@@ -583,6 +585,7 @@
 	# Install current database files
 	#
 	@if [ -d $(FOODB) ]; then \
+		[ -d $(FOODB)/$$dir/ ] || install -d -m 755 $(FOODB)/$$dir/; \
 	    for dir in driver printer opt; do \
 		echo install -m 644 foomatic-db/$$dir/*.xml $(FOODB)/$$dir/; \
 		$(INSTALL) -c -m 644 foomatic-db/$$dir/*.xml $(FOODB)/$$dir/; \
@@ -601,10 +604,10 @@
 	#
 	# Clear foomatic cache and rebuild database if needed
 	#
-	rm -rf /var/cache/foomatic/*/*
-	rm -f /var/cache/foomatic/printconf.pickle
-	if [ -d /var/cache/foomatic/compiled ]; then \
-	    cd /var/cache/foomatic/compiled; \
+	rm -rf $(DESTDIR)/var/cache/foomatic/*/*
+	rm -f $(DESTDIR)/var/cache/foomatic/printconf.pickle
+	if [ -d $(DESTDIR)/var/cache/foomatic/compiled ]; then \
+	    cd $(DESTDIR)/var/cache/foomatic/compiled; \
 	    foomatic-combo-xml -O >overview.xml; \
 	fi
 
@@ -736,13 +739,6 @@
 	    done; \
 	    ppdmgr -u; \
 	elif [ -d $(PPD) ]; then \
-	    find $(PPD) -name '*foo2zjs*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2hp*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2xqx*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2lava*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2qpdl*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2slx*' | xargs rm -rf; \
-	    find $(PPD) -name '*foo2hiperc*' | xargs rm -rf; \
             [ -d $(PPD)/foo2zjs ] || mkdir $(PPD)/foo2zjs; \
 	    cd PPD; \
 	    for ppd in *.ppd; do \
@@ -789,10 +785,8 @@
 	$(INSTALL) -c -m 755 hplj10xx_gui.tcl $(SHAREZJS)
 	
 
-USBDIR=/etc/hotplug/usb
-UDEVDIR=/etc/udev/rules.d
-RULES=hplj10xx.rules
-install-hotplug: install-hotplug-test install-hotplug-prog
+USBDIR=$(DESTDIR)/etc/hotplug/usb
+install-hotplug: install-hotplug-test install-udev
 
 install-hotplug-test:
 	#
@@ -810,9 +804,6 @@
 	#
 
 install-hotplug-prog:
-	if [ -d $(UDEVDIR) ]; then \
-	    $(INSTALL) -c -m 644 $(RULES) $(UDEVDIR)/11-$(RULES); \
-	fi
 	[ -d $(USBDIR) ] || $(INSTALL) -d -m 755 $(USBDIR)/
 	$(INSTALL) -c -m 755 hplj1000 $(USBDIR)/
 	ln -sf $(USBDIR)/hplj1000 $(USBDIR)/hplj1005
@@ -823,15 +814,12 @@
 	ln -sf $(USBDIR)/hplj1000 $(USBDIR)/hpljP1007
 	ln -sf $(USBDIR)/hplj1000 $(USBDIR)/hpljP1008
 	ln -sf $(USBDIR)/hplj1000 $(USBDIR)/hpljP1505
-	$(USBDIR)/hplj1000 install-usermap
-	$(USBDIR)/hplj1005 install-usermap
-	$(USBDIR)/hplj1018 install-usermap
-	$(USBDIR)/hplj1020 install-usermap
-	$(USBDIR)/hpljP1005 install-usermap
-	$(USBDIR)/hpljP1006 install-usermap
-	$(USBDIR)/hpljP1007 install-usermap
-	$(USBDIR)/hpljP1008 install-usermap
-	$(USBDIR)/hpljP1505 install-usermap
+	install -c -m 644 hplj.usermap $(USBDIR)/
+UDEVDIR=$(DESTDIR)/etc/udev/rules.d
+RULES=hplj10xx.rules
+install-udev:
+	[ -d $(UDEVDIR) ] || install -d -m 755 $(UDEVDIR)/
+	install -c -m 644 $(RULES) $(UDEVDIR)/11-$(RULES)
 
 cups:	FRC
 	if [ -x /etc/init.d/cups ]; then \
@@ -1223,7 +1211,6 @@
 	$(INSTALL) -c -m 644 README $(DOCDIR)
 	$(INSTALL) -c -m 644 ChangeLog $(DOCDIR)
 
-GROFF=/usr/local/test/bin/groff
 GROFF=groff
 manual.pdf: $(MANPAGES)
 	-$(GROFF) -t -man $(MANPAGES) | ps2pdf - $@
