diff -ur ispell-3.3.02.orig/correct.c ispell-3.3.02/correct.c
--- ispell-3.3.02.orig/correct.c	2009-08-06 18:28:32.000000000 +0300
+++ ispell-3.3.02/correct.c	2009-08-06 18:28:57.000000000 +0300
@@ -247,7 +247,7 @@
 		  struct flagent * sufent,
 		  ichar_t savearea[MAX_CAPS][INPUTWORDLEN + MAXAFFIXLEN],
 		  int * nsaved));
-static char *	getline P ((char * buf, int bufsize));
+static char *	get_line P ((char * buf, int bufsize));
 void		askmode P ((void));
 void		copyout P ((unsigned char ** cc, int cnt));
 static void	lookharder P ((unsigned char * string));
@@ -573,7 +573,7 @@
 
 		imove (li - 1, 0);
 		(void) putchar ('!');
-		if (getline ((char *) buf, sizeof buf) == NULL)
+		if (get_line ((char *) buf, sizeof buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    ierase ();
@@ -598,7 +598,7 @@
 		    (void) printf ("%s ", CORR_C_READONLY);
 		    }
 		(void) printf (CORR_C_REPLACE_WITH);
-		if (getline ((char *) ctok, ctokl) == NULL)
+		if (get_line ((char *) ctok, ctokl) == NULL)
 		    {
 		    (void) putchar (7);
 		    /* Put it back */
@@ -666,7 +666,7 @@
 		unsigned char	buf[100];
 		imove (li - 1, 0);
 		(void) printf (CORR_C_LOOKUP_PROMPT);
-		if (getline ((char *) buf, sizeof buf) == NULL)
+		if (get_line ((char *) buf, sizeof buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    ierase ();
@@ -1585,7 +1585,7 @@
     return;
     }
 
-static char * getline (s, len)
+static char * get_line (s, len)
     register char *	s;
     register int	len;
     {
